tests/sphere_test.cpp

```diff
@@ -0,0 +1,100 @@
+#include "obj/objects.h"
+#include "gtest/gtest.h"
+
+using core::Ray;
+using core::RayHitData;
+using core::Vec3;
+using objects::Sphere;
+
+const float kSkipFreq = 1000;
+const Vec3 kVecZero(0, 0, 0);
+
+TEST(SphereCollisionTest, RayHitFromOutsideSphere) {
+  Sphere sphere(kVecZero, 1);
+  RayHitData hitData;
+
+  Ray alongYAxis(Vec3(0, -4, 0), Vec3(0, 1, 0));
+  ASSERT_TRUE(sphere.hitObject(alongYAxis, kSkipFreq, &hitData));
+  ASSERT_FLOAT_EQ(alongYAxis.origin().magnitude() - sphere.getRadius(),
+                  hitData.time);
+
+  Ray alongYAxisNoHit(Vec3(0, 4, 0), Vec3(0, 1, 0));
```

Comment:
It is clear that it is "noHit" from the ASSERT_FALSE

Why not describe the relation to the sphere itself, like `alongYAxisAwayFromSphere`?